lib/jellyfish_web/peer_socket.ex

```diff
@@ -75,57 +76,37 @@ defmodule JellyfishWeb.PeerSocket do
     end
   end
 
-  def handle_in({encoded_message, [opcode: :text]}, state) do
-    case Jason.decode(encoded_message) do
-      {:ok, %{"type" => "mediaEvent", "data" => data}} ->
-        send(state.room_pid, {:media_event, state.peer_id, data})
+  @impl true
+  def handle_in({encoded_message, [opcode: :binary]}, state) do
+    case PeerMessage.decode(encoded_message) do
+      %PeerMessage{content: {:media_event, %MediaEvent{data: data}}} ->
+        Room.pass_media_event(state.room_id, state.peer_id, data)
 
-      {:error, %Jason.DecodeError{}} ->
+      _other ->
         Logger.warn("""
-        Failed to decode message from peer #{inspect(state.peer_id)}, \
+        Received unexpected message from #{inspect(state.peer_id)}, \
         room: #{inspect(state.room_id)}
```

Comment:
Let's also inspect this message and the previous log error was correct, wasn't it?  I mean, "Failed to decode message from peer"

Reply:
AFAIK `PeerMessage.decode` will raise when failed to decode the message, so I think `unexpected message` make sense here. I'll add the message to log ofc.